packages/react-components/src/components/DevicePermissions/DomainPermissionsScaffolding.tsx

```diff
@@ -0,0 +1,164 @@
+  IButtonStyles,
+  mergeStyleSets
+} from '@fluentui/react';
+import { _formatString, _pxToRem } from '@internal/acs-ui-common';
+import {
+  iconContainerStyles,
+  iconPrimaryStyles,
+  linkTextStyles,
+  primaryButtonStyles,
+  primaryTextStyles,
+  secondaryTextStyles,
+  sparkleIconBackdropStyles,
+  textContainerStyles
+} from '../styles/DomainPermissions.styles';
+import { isValidString } from '../utils';
+import { BaseCustomStyles } from '../../types';
+
+/**
+ * @private
+ * Props for DomainPermissions component.
+ */
+export interface DomainPermissionsContainerProps {
+  /**
+   * Name of application calling experience is in.
+   */
+  appName?: string;
+  /**
+   * Name of icon to be used for the camera icon.
+   * If this is not provided the icon will not be shown.
+   */
+  cameraIconName?: string;
+  /**
+   * Name of icon to be used for the sparkle icon.
+   * If this is not provided the icon will not be shown.
+   */
+  connectorIconName?: string;
+  /**
+   * Name of icon to be used for the mic icon.
+   * If this is not provided the icon will not be shown.
+   */
+  microphoneIconName?: string;
+  /**
+   * Action to be taken by the more help link. Possible to send to external page or show other modal.
+   * If this is not provided the button will not be shown.
+   */
+  onTroubleshootingClick?: () => void;
+  /**
+   * Action that is taken when the user clicks an optional primary button.
+   * If this is not provided the button will not be shown.
+   */
+  onPrimaryButtonClick?: () => void;
+  /**
+   * Localization strings for DomainPermissions component.
+   */
+  strings?: DomainPermissionsStrings;
+  /**
+   * Styles for DomainPermissions component.
+   */
+  styles?: DomainPermissionsStyles;
+}
+
+/**
+ * @beta
+ * Strings for DomainPermissions component
+ */
+export type DomainPermissionsStrings = {
+  /**
+   * Main text string.
+   */
+  primaryText?: string;
+  /**
+   * Subtext string.
+   */
+  secondaryText?: string;
+  /**
+   * More help link string.
+   */
+  linkText?: string;
+  /**
+   * Primary button text string.
+   */
+  primaryButtonText?: string;
+  /**
+   * Aria label describing the content of the container
+   */
+  ariaLabel?: string;
+};
+
+/**
+ * Fluent styles for {@link DomainPermissionsContainer}.
+ *
+ * @beta
+ */
+export interface DomainPermissionsStyles extends BaseCustomStyles {
+  /** Styles for the primary button. */
+  primaryButton?: IButtonStyles;
+  /** Styles for the help troubleshooting link text. */
+  troubleshootingLink?: ILinkStyles;
+}
+
+/** @private */
+export const DomainPermissionsContainer = (props: DomainPermissionsContainerProps): JSX.Element => {
+  const { appName, onTroubleshootingClick, onPrimaryButtonClick, strings } = props;
+  const theme = useTheme();
+  return (
+    <Stack style={{ padding: '2rem', maxWidth: '25.375rem' }} aria-label={strings?.ariaLabel}>
+      <Stack horizontal style={{ paddingBottom: '1rem' }} horizontalAlign={'space-between'}>
```

Comment:
nit: this should have `props.cameraIconName || props.connectorIconName || props.microphoneIconName`